### ISSUE TITLE
Esercizio 3

### ISSUE THREAD

**Issue by matypist** [sapienzastudentsnetwork/mmi2122#52]
![image](https://user-images.githubusercontent.com/69389461/152314840-f57d3dc5-ad88-4de9-a6d4-83e5ac8b9bb8.png)

**Comment by LuginiAndrea** [comment]
Prendiamo $Z = \{f\left(x\right), \forall x \in X}$.
Definiamo ora la funzione $h: Z \rightarrow Y$ come la funzione identità (è quindi implicato che sia iniettiva), e definiamo la funzione $g: X \rightarrow Z$ come la funzione $f$ (è quindi implicato che sia suriettiva in quanto $Z = f\left(X\right)$.
Inoltre essendo $h$ definita come la funzione identità e $g$ come la funzione $f$ è implicato che $h \circ g = f$.
Abbiamo quindi dimostrato che esiste l'insieme $Z$ che ci permette di rispettare le premesse dell'ipotesi

**Comment by FrascaHoryzont** [comment]
Vero, perché se considero f come funzione da X alla sua immagine via f, associo ad ogni elemento di X un elemento di Y.
E se considero una funzione composta formata da hog ovvero h(g(x)), ottengo una funzione che svolge lo stesso compito di f ovvero che associa ad ogni elemento dell’insieme di X un elemento dell’insieme Y seppur passando attraverso l’insieme Z. Quindi le due funzioni f e hog sono identiche ed è quindi possibile ogni funzione f come composizione di una funzione surriettiva e una funzione iniettiva. Allego con la risposta la dimostrazione tramite schema.
![image](https://user-images.githubusercontent.com/100992294/212096850-966a5f6e-c841-4362-8b08-e93f1684c3b4.jpg)

**Comment by PabDotExe** [comment]
Se seguiamo il suggerimento e quindi consideriamo **f(X)** allora:
**f(X) = (h o g)** 
allora svolgiamola inserendo come argomento di **g** l'insieme **X**:
**(h o g) = h(g(X))** 
come sappiamo **g: X -> Z** quindi:
**h(g(X))** =  **h(Z)** 
come prima **h: Z -> Y** quindi:
**h(Z) = Y**
Quindi possiamo concludere abbuiamo dimostrato che **f(X) = (h o g) = Y** perché **f(X) = Y** .
